cvastrophoto/rops/denoise/diffusion.py

```diff
@@ -0,0 +1,160 @@
+# cython: infer_types=True
+
+import PIL
+import math
+import numpy
+import cython
+import logging
+from skimage import color, measure
+from ..base import PerChannelRop
+if not cython.compiled:
+    globals().update(dict(fabs=abs))
+
+
+logger = logging.getLogger(__name__)
+
+class DiffusionRop(PerChannelRop):
+    def __init__(self,raw=None,T=0.0025,L=1.0/180,R=2,thr=3,steps=20,dt=1.0/7):
+        super(DiffusionRop,self).__init__(raw)
+        self.T = T
+        self.L = L
+        self.R = R
+        self.thr = thr
+        self.dt = dt
+        self.steps = steps
+    def process_channel(self,data,detected=None): 
+        accelerated=False
+        denoiser = Denoise(self.L,accelerated)
+        return denoiser.modifiedPMdiffusion(data,self.steps,self.R,self.T,self.thr,self.dt)
+ 
+     
+@cython.cfunc
+@cython.locals(gradient=cython.double, L=cython.double, g2=cython.double)
+@cython.returns(cython.double)
+@cython.nogil
+def g(L,gradient):
+    g2 = fabs(gradient)/L
+    return (1.0/(1.0+ g2*g2))
+
+@cython.cclass
+class Denoise(object):
+    cython.declare(
+        L=cython.double,
+        accelerated=cython.bint,
+    )
+
+    def __init__(self,L,accelerated=False):
+        self.L= L
+        self.accelerated = accelerated
+        
+        
+    def modifiedPMdiffusion(self,noisy,steps, nradius,T,thr,dt):
+        ut=noisy
+        ssim = numpy.zeros(shape=steps, dtype=numpy.float32)
+        psnr = numpy.zeros(shape=steps, dtype=numpy.float32)
+        
+
+        for i in range(steps):
+            maskt, evolvedut= self.binarymask(ut,nradius,T,thr)
+            ut, changed = self.iteration(maskt,evolvedut,dt)
+            #psnr[i]= measure.compare_psnr(evolvedut,ut)
+            #ssim[i] = measure.compare_ssim(evolvedut,ut)
```

Comment:
Remove commented code (if it's experimental code, you can re-add it later in your branch, but the PR should be free of commented code)